test-analysis-core/src/main/kotlin/com/saveourtool/save/test/analysis/api/metrics/TestMetrics.kt

```diff
@@ -0,0 +1,10 @@
+package com.saveourtool.save.test.analysis.api.metrics
+
+import com.saveourtool.save.test.analysis.api.TestStatisticsStorage
+
+/**
+ * Scalar test metrics returned by [TestStatisticsStorage.getTestMetrics].
+ *
+ * @see TestStatisticsStorage.getTestMetrics
+ */
+sealed interface TestMetrics
```

Comment:
Should these classes (`TestMetrics` and `AnalysisResults` and their descendants) be located in `save-cloud-common` from the very beginning since we'll probably be storing this information in the database in the future?

Reply:
Both `TestMetrics` and `AnalysisResults` are actually denormalized data.

So my initial idea was to read the test run history in the background as the back-end service starts, and then keep continously updating the data as new tests are run.

Reply:
Yes, I see that it's denormalized, but are you suggesting to keep statistics in memory only? Even if these algorithms require the whole series of execution to calculate or update, wouldn't it slow the things too much when statistics are only being read?

Reply:
> ... but are you suggesting to keep statistics in memory only

Well, _TeamCity_ does exactly that, keeping the last N runs of hundreds of thousands of tests in-memory.

The idea is to keep only a sub-volume of the whole statistical data available (the _sliding window_ discards older test runs), but maintain fast access to this data (when multiple clients are requesting analysis for tests X, Y, Z and so on).

There's the only limitation: your data structures should be as small as possible to avoid OOM.

If the volume of the underlying test run history is not that large, and the number of concurrent users is low, indeed we can read directly from the database, but at this (early prototype) stage the final decision is actually not that important.

We can implement initial support from the front-end and the back-end now, play for a while, and add extra persistent entities later.